src/Compilers/CSharp/Portable/Symbols/Compilation_WellKnownMembers.cs

```diff
@@ -42,12 +43,24 @@ internal EmbeddableAttributes GetNeedsGeneratedAttributes()
             return (EmbeddableAttributes)_needsGeneratedAttributes;
         }
 
-        internal void SetNeedsGeneratedAttributes(EmbeddableAttributes attributes)
+        private void SetNeedsGeneratedAttributes(EmbeddableAttributes attributes)
         {
             Debug.Assert(!_needsGeneratedAttributes_IsFrozen);
             ThreadSafeFlagOperations.Set(ref _needsGeneratedAttributes, (int)attributes);
         }
 
+        internal bool GetUsesNullableAttributes()
+        {
+            _needsGeneratedAttributes_IsFrozen = true;
+            return _usesNullableAttributes;
+        }
+
+        private void SetUsesNullableAttributes()
+        {
+            Debug.Assert(!_needsGeneratedAttributes_IsFrozen);
+            _usesNullableAttributes = true;
```

Comment:
>_usesNullableAttributes = true; [](start = 12, length = 31)

just to confirm, we don't need any special operation to ensure this set is atomic, correct?

Reply:
Yes, reading and writing a `bool` is atomic.

---
In reply to: [301355085](https://github.com/dotnet/roslyn/pull/37019#discussion_r301355085) [](ancestors = 301355085)